controller/control/controller.go

```diff
@@ -6,14 +6,15 @@ import (
 
 	"github.com/containerd/console"
 	controllerapi "github.com/docker/buildx/controller/pb"
+	"github.com/moby/buildkit/client"
 )
 
 type BuildxController interface {
-	Invoke(ctx context.Context, ref string, options controllerapi.ContainerConfig, ioIn io.ReadCloser, ioOut io.WriteCloser, ioErr io.WriteCloser) error
-	Build(ctx context.Context, options controllerapi.BuildOptions, in io.ReadCloser, w io.Writer, out console.File, progressMode string) (ref string, err error)
+	Build(ctx context.Context, options controllerapi.BuildOptions, in io.ReadCloser, w io.Writer, out console.File, progressMode string) (ref string, resp *client.SolveResponse, err error)
```

Comment:
Why is progressmode part of controller interface.

Maybe this is semantics. Is `Controller` interface that different implementation implement (eg. direct/remote)? Or is it a high-level client interface a caller would use to invoke builds. I don't think it is the same thing.

Reply:
cc @ktock, do you have additional context for this?

Reply:
This is because remote/local controllers have different implementations for status handling (local: printed by `(controller/build).RunBuild` vs remote: printed by `(controller/remote).Client.Build()`). But I think we can now refactor this API to forward `chan *client.SolveStatus` instead of passing `progressMode` via that arg.

Reply:
Opened https://github.com/docker/buildx/pull/1671